workspace/__lib__/xod/datetime/pad-with-zeroes/patch.cpp

```diff
@@ -0,0 +1,33 @@
+
+struct State {
+    char str[16];
+    CStringView view;
+    State()
+        : view(str) {}
+};
+
+// clang-format off
+{{ GENERATED_CODE }}
+// clang-format on
+
+void evaluate(Context ctx) {
+    auto state = getState(ctx);
+
+    auto num = abs(getValue<input_IN>(ctx));
+    uint8_t lenFull = min(15, getValue<input_W>(ctx));
+
+    char strNum[16];
+    dtostrf(num, 0, 0, strNum);
+
+    CStringView numView(strNum);
+    uint8_t lenStr = length(XString(&numView));
+
+    if (lenStr >= lenFull) {
+        strcpy(state->str, strNum);
+    } else {
```

Comment:
I think the `if` clause is no longer required at all. The `else` branch will do the job for any case. Please, confirm.